x/programs/rust/wasmlanche_sdk/src/memory.rs

```diff
@@ -94,6 +94,10 @@ pub extern "C" fn alloc(len: usize) -> *mut u8 {
     let mut buf = Vec::with_capacity(len);
     // take a mutable pointer to the buffer
     let ptr = buf.as_mut_ptr();
+    // ensure memory pointer is fits in an i64
+    // to avoid potential issues 
+    // when passing across wasm boundary
+    assert!(ptr as u64 <= i64::MAX as u64);
```

Comment:
wrote an example to see what happens if we return > `i64::MAX` and wasm will just round to `i64::MAX`. Still an important check IMO since this ensures the ptr's we get back are correct.